### PR TITLE
feat: rss3 api update

### PR DESCRIPTION
**Pull request by Randolph314**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-providers/src/types.ts [hunk 1]
```diff
@@ -129,8 +217,8 @@ export namespace RSS3BaseAPI {
         createRSS3(address: string): RSS3
         getFileData<T>(rss3: RSS3, address: string, key: string): Promise<T | undefined>
         setFileData<T>(rss3: RSS3, address: string, key: string, data: T): Promise<T>
-        getDonations(address: string): Promise<GeneralAssetResponse | undefined>
-        getFootprints(address: string): Promise<GeneralAssetResponse | undefined>
+        getDonations(address: string): Promise<Donations[] | undefined>
+        getFootprints(address: string): Promise<Footprints[] | undefined>
```
Comment (guanbinrui) [inline packages/web3-providers/src/types.ts]:
```suggestion
        getDonations(address: string): Promise<Donation[] | undefined>
        getFootprints(address: string): Promise<Footprint[] | undefined>
```

File: packages/web3-providers/src/types.ts [hunk 2]
```diff
@@ -113,6 +113,94 @@ export namespace RSS3BaseAPI {
         name: string
     }
 
+    export interface NFTContract {
+        address: string
+        name: string
+        symbol: string
+    }
+
+    export interface NFT_traits {
```
Comment (guanbinrui) [inline packages/web3-providers/src/types.ts]:
```suggestion
    export interface NFT_Trait {
```

File: packages/web3-providers/src/types.ts [hunk 3]
```diff
@@ -113,6 +113,94 @@ export namespace RSS3BaseAPI {
         name: string
     }
 
+    export interface NFTContract {
```
Comment (guanbinrui) [inline packages/web3-providers/src/types.ts]:
```suggestion
    export interface NFT_Contract {
```

File: packages/web3-providers/src/types.ts [hunk 4]
```diff
@@ -113,6 +113,94 @@ export namespace RSS3BaseAPI {
...
+        id: string
+        detail: NFT_Type
+    }
+
+    export interface DonationTx {
+        adminAddr: string
+        amount: string
+        approach: string
+        donor: string
+        formatedAmount: string
```
Comment (UncleBill) [inline packages/web3-providers/src/types.ts:152]: `formated` is a typo, is it defined by RSS3? If so, then we can ignore it.

Comment (Randolph314) [reply packages/web3-providers/src/types.ts:152]: yes, RSS3 api defined it

File: packages/web3-providers/src/types.ts [hunk 5]
```diff
@@ -113,6 +113,94 @@ export namespace RSS3BaseAPI {
         name: string
     }
 
+    export interface NFT_Contract {
+        address: string
+        name: string
+        symbol: string
+    }
+
+    export interface NFT_Traits {
```
Comment (guanbinrui) [inline packages/web3-providers/src/types.ts]:
```suggestion
    export interface NFT_Trait {
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**